diff -rupN devicetree.orig/pcw.dtsi devicetree/pcw.dtsi
--- devicetree.orig/pcw.dtsi
+++ devicetree/pcw.dtsi
@@ -13,9 +13,11 @@
 	};
 };
 &gem0 {
+	local-mac-address = [00 0a 35 00 00 00];
 	phy-mode = "rgmii-id";
 	status = "okay";
 	xlnx,ptp-enet-clock = <0x69f6bcb>;
+	phy-handle = <&phy0>;
 };
 &gpio0 {
 	emio-gpio-width = <64>;
@@ -35,11 +37,38 @@
 	num_interrupts = <96>;
 };
 &qspi {
+	status = "okay";
 	is-dual = <0>;
 	num-cs = <1>;
-	spi-rx-bus-width = <4>;
-	spi-tx-bus-width = <4>;
-	status = "okay";
+	flash@0 {
+		compatible = "spansion,s25fl512s";
+		reg = <0x0>;
+		spi-tx-bus-width = <1>;
+		spi-rx-bus-width = <4>;
+		spi-max-frequency = <50000000>;
+		#address-cells = <1>;
+		#size-cells = <1>;
+		partition@qspi-fsbl-uboot {
+			label = "qspi-fsbl-uboot";
+			reg = <0x0 0x100000>;
+		};
+		partition@qspi-linux {
+			label = "qspi-linux";
+			reg = <0x100000 0x500000>;
+		};
+		partition@qspi-device-tree {
+			label = "qspi-device-tree";
+			reg = <0x600000 0x100000>;
+		};
+		partition@qspi-bitstream {
+			label = "qspi-bitstream";
+			reg = <0x700000 0x100000>;
+		};
+		partition@qspi-rootfs {
+			label = "qspi-rootfs";
+			reg = <0x800000 0x3800000>;
+		};
+	};
 };
 &sdhci0 {
 	status = "okay";
@@ -51,11 +80,21 @@
 	is-decoded-cs = <0>;
 	num-cs = <3>;
 	status = "okay";
+	spidev@0 {
+		compatible = "spidev";
+		reg = <0x0>;
+		spi-max-frequency = <10000000>;
+	};
 };
 &spi1 {
 	is-decoded-cs = <0>;
 	num-cs = <3>;
 	status = "okay";
+	spidev@0 {
+		compatible = "spidev";
+		reg = <0x0>;
+		spi-max-frequency = <10000000>;
+	};
 };
 &uart0 {
 	device_type = "serial";
@@ -63,7 +102,8 @@
 	status = "okay";
 };
 &usb0 {
-	phy_type = "ulpi";
+	dr_mode = "host";
+	usb-phy = <&usb_phy0>;
 	status = "okay";
 };
 &clkc {
diff -rupN devicetree.orig/system-top.dts devicetree/system-top.dts
--- devicetree.orig/system-top.dts
+++ devicetree/system-top.dts
@@ -14,6 +14,10 @@
 		bootargs = "console=ttyPS0,115200 root=/dev/mmcblk0p2 ro rootfstype=ext4 earlyprintk rootwait earlycon";
 		stdout-path = "serial0:115200n8";
 	};
+	usb_phy0: phy1 {
+		#phy-cells = <0>;
+		compatible = "usb-nop-xceiv";
+	};
 	aliases {
 		ethernet0 = &gem0;
 		i2c0 = &i2c0;
@@ -25,6 +29,19 @@
 	};
 	memory {
 		device_type = "memory";
-		reg = <0x0 0x20000000>;
+		reg = <0x0 0x1E000000>;
 	};
+	mdio {
+		#address-cells = <1>;
+		#size-cells = <0>;
+		phy0: ethernet-phy@3 {
+			compatible = "micrel,ksz9031";
+			device_type = "ethernet-phy";
+			rxc-skew-ps = <1860>;
+			rxdv-skew-ps = <420>;
+			txc-skew-ps = <1860>;
+			txen-skew-ps = <420>;
+			reg = <3>;
+		};
+   };
 };
