diff -rupN devicetree.orig/pcw.dtsi devicetree/pcw.dtsi
--- devicetree.orig/pcw.dtsi
+++ devicetree/pcw.dtsi
@@ -13,9 +13,11 @@
 	};
 };
 &gem0 {
+	local-mac-address = [00 0a 35 00 00 00];
 	phy-mode = "rgmii-id";
 	status = "okay";
 	xlnx,ptp-enet-clock = <0x69f6bcb>;
+	phy-handle = <&phy0>;
 };
 &gpio0 {
 	emio-gpio-width = <64>;
@@ -49,11 +51,21 @@
 	is-decoded-cs = <0>;
 	num-cs = <3>;
 	status = "okay";
+	spidev@0 {
+		compatible = "spidev";
+		reg = <0x0>;
+		spi-max-frequency = <10000000>;
+	};
 };
 &spi1 {
 	is-decoded-cs = <0>;
 	num-cs = <3>;
 	status = "okay";
+	spidev@0 {
+		compatible = "spidev";
+		reg = <0x0>;
+		spi-max-frequency = <10000000>;
+	};
 };
 &uart0 {
 	device_type = "serial";
@@ -62,8 +74,9 @@
 };
 &usb0 {
 	dr_mode = "host";
-	phy_type = "ulpi";
+	usb-phy = <&usb_phy0>;
 	status = "okay";
+	xlnx,phy-reset-gpio = <&gpio0 48 0>;
 };
 &clkc {
 	fclk-enable = <0x1>;
diff -rupN devicetree.orig/system-top.dts devicetree/system-top.dts
--- devicetree.orig/system-top.dts
+++ devicetree/system-top.dts
@@ -14,6 +14,13 @@
 		bootargs = "console=ttyPS0,115200 root=/dev/mmcblk0p2 ro rootfstype=ext4 earlyprintk rootwait earlycon";
 		stdout-path = "serial0:115200n8";
 	};
+	usb_phy0: phy0 {
+		#phy-cells = <0>;
+		compatible = "ulpi-phy";
+		reg = <0xe0002000 0x1000>;
+		view-port = <0x0170>;
+		drv-vbus;
+	};
 	aliases {
 		ethernet0 = &gem0;
 		serial0 = &uart0;
@@ -23,8 +30,21 @@
 	};
 	memory {
 		device_type = "memory";
-		reg = <0x0 0x20000000>;
+		reg = <0x0 0x1E000000>;
 	};
 	cpus {
 	};
+	mdio {
+		#address-cells = <1>;
+		#size-cells = <0>;
+		phy0: ethernet-phy@3 {
+			compatible = "micrel,ksz9031";
+			device_type = "ethernet-phy";
+			rxc-skew-ps = <1860>;
+			rxdv-skew-ps = <420>;
+			txc-skew-ps = <1860>;
+			txen-skew-ps = <420>;
+			reg = <3>;
+		};
+	};
 };
