diff -rupN devicetree.orig/pcw.dtsi devicetree/pcw.dtsi
--- devicetree.orig/pcw.dtsi
+++ devicetree/pcw.dtsi
@@ -25,6 +25,10 @@
 &i2c0 {
 	clock-frequency = <400000>;
 	status = "okay";
+	eep@50 {
+		compatible = "24c64";
+		reg = <0x50>;
+	};
 };
 &intc {
 	num_cpus = <2>;
@@ -47,11 +51,21 @@
 	is-decoded-cs = <0>;
 	num-cs = <3>;
 	status = "okay";
+	spidev@0 {
+		compatible = "spidev";
+		reg = <0x0>;
+		spi-max-frequency = <10000000>;
+	};
 };
 &spi1 {
 	is-decoded-cs = <0>;
 	num-cs = <1>;
 	status = "okay";
+	spidev@0 {
+		compatible = "spidev";
+		reg = <0x0>;
+		spi-max-frequency = <10000000>;
+	};
 };
 &uart0 {
 	device_type = "serial";
@@ -64,7 +78,8 @@
 	status = "okay";
 };
 &usb0 {
-	phy_type = "ulpi";
+	dr_mode = "host";
+	phy_type = <&usb_phy0>;
 	status = "okay";
 	usb-reset = <&gpio0 48 0>;
 };
diff -rupN devicetree.orig/system-top.dts devicetree/system-top.dts
--- devicetree.orig/system-top.dts
+++ devicetree/system-top.dts
@@ -14,6 +14,10 @@
 		bootargs = "console=ttyPS0,115200 root=/dev/mmcblk0p2 ro rootfstype=ext4 earlyprintk rootwait earlycon";
 		stdout-path = "serial0:115200n8";
 	};
+	usb_phy0: phy1 {
+		#phy-cells = <0>;
+		compatible = "usb-nop-xceiv";
+	};
 	aliases {
 		ethernet0 = &gem0;
 		i2c0 = &i2c0;
@@ -25,6 +29,6 @@
 	};
 	memory {
 		device_type = "memory";
-		reg = <0x0 0x20000000>;
+		reg = <0x0 0x1E000000>;
 	};
 };
