diff -rupN devicetree.orig/pcw.dtsi devicetree/pcw.dtsi
--- devicetree.orig/pcw.dtsi	2018-06-28 16:53:52.592601519 +0200
+++ devicetree/pcw.dtsi	2018-06-28 16:53:52.596601519 +0200
@@ -51,6 +51,11 @@
 	is-decoded-cs = <0>;
 	num-cs = <3>;
 	status = "okay";
+	spidev@0 {
+		compatible = "spidev";
+		reg = <0x0>;
+		spi-max-frequency = <10000000>;
+	};
 };
 &uart1 {
 	device_type = "serial";
diff -rupN devicetree.orig/system.dts devicetree/system.dts
--- devicetree.orig/system.dts	2018-06-28 16:53:52.592601519 +0200
+++ devicetree/system.dts	2018-07-17 14:59:20.537522242 +0200
@@ -6,5 +6,5 @@
 
 
 &gem0 {
-	local-mac-address = [00 0a 35 00 00 00];
+	local-mac-address = [00 0a 35 00 00 01];
 };
diff -rupN devicetree.orig/system-top.dts devicetree/system-top.dts
--- devicetree.orig/system-top.dts	2018-06-28 16:53:52.592601519 +0200
+++ devicetree/system-top.dts	2018-06-28 16:53:52.596601519 +0200
@@ -22,7 +22,7 @@
 	};
 	memory {
 		device_type = "memory";
-		reg = <0x0 0x40000000>;
+		reg = <0x0 0x30000000>;
 	};
 	cpus {
 	};
