diff -rupN devicetree.orig/pcw.dtsi devicetree/pcw.dtsi
--- devicetree.orig/pcw.dtsi
+++ devicetree/pcw.dtsi
@@ -47,6 +47,11 @@
 	is-decoded-cs = <0>;
 	num-cs = <3>;
 	status = "okay";
+	spidev@0 {
+		compatible = "spidev";
+		reg = <0x0>;
+		spi-max-frequency = <10000000>;
+	};
 };
 &uart1 {
 	device_type = "serial";
@@ -54,7 +59,7 @@
 	status = "okay";
 };
 &usb0 {
-	phy_type = "ulpi";
+	usb-phy = <&usb_phy0>;
 	status = "okay";
 	usb-reset = <&gpio0 7 0>;
 };
diff -rupN devicetree.orig/system-top.dts devicetree/system-top.dts
--- devicetree.orig/system-top.dts
+++ devicetree/system-top.dts
@@ -14,15 +14,22 @@
 		bootargs = "console=ttyPS0,115200 root=/dev/mmcblk0p2 ro rootfstype=ext4 earlyprintk rootwait earlycon";
 		stdout-path = "serial0:115200n8";
 	};
+	usb_phy0: phy0 {
+		#phy-cells = <0>;
+		compatible = "ulpi-phy";
+		reg = <0xe0002000 0x1000>;
+		view-port = <0x0170>;
+		drv-vbus;
+	};
 	aliases {
 		ethernet0 = &gem0;
 		i2c0 = &i2c0;
 		serial0 = &uart1;
 		spi0 = &qspi;
-		spi1 = &spi0;
+		spi2 = &spi0;
 	};
 	memory {
 		device_type = "memory";
-		reg = <0x0 0x20000000>;
+		reg = <0x0 0x1E000000>;
 	};
 };